src/__tests__/HeaderContainer.test.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import 'jest-styled-components';
+import renderer from 'react-test-renderer';
+import '@testing-library/jest-dom/extend-expect';
+import Header from '../components/Header/Header.component';
+import MenuIcon from '../components/Header/MenuButton.component';
+import SessionButton from '../components/Header/SessionButton.component';
+import ThemeSelector from '../components/Header/ThemeSelector.component';
+import SearchBar from '../components/Header/SearchBar.component';
+import HeaderContainer from '../components/Header/HeaderContainer.component';
+
+describe('validate header container works properly', () => {
+    test('render header contains elements', () => {
+        const headerC = renderer.create(<HeaderContainer />).toJSON();
+        expect(headerC.children.length).toBe(4);
+    });
+});
+
+describe('validate header works properly', () => {
+  test('render header is working', () => {
+    const hdr = renderer.create(<Header />).toJSON();
+        expect(hdr).toMatchSnapshot()
```

Comment:
Instead of using snapshot for testing render output, check for assertions on rendered content.